internal/configuration/reader.go

```diff
@@ -38,6 +46,7 @@ func Read(configPath string) (*schema.Configuration, []error) {
 	viper.Unmarshal(&configuration) //nolint:errcheck // TODO: Legacy code, consider refactoring time permitting.
 
 	val := schema.NewStructValidator()
+	validator.ValidateSecrets(&configuration, val, viper.GetViper())
 	validator.Validate(&configuration, val)
```

Comment:
I'd rename Validate into ValidateConfiguration to emphasize the distinction between the two public methods.